api/kogito-api/src/main/java/org/kie/kogito/Models.java

```
@@ -0,0 +1,113 @@
+                if (e.getKey().startsWith("v$")) {
+                    k = k.substring(2);
+                }
+                map.put(k, e.getValue().getReadMethod().invoke(m));
+            }
+            return map;
+        } catch (IntrospectionException | IllegalAccessException | InvocationTargetException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    public static <T> T fromMap(T m, Map<String, Object> map) {
+        try {
+            BeanInfo beanInfo = Introspector.getBeanInfo(m.getClass());
+            Map<String, PropertyDescriptor> descriptors = descriptorMap(beanInfo);
+
+            for (Map.Entry<String, PropertyDescriptor> e : descriptors.entrySet()) {
+                String k = e.getKey();
+                if (e.getKey().startsWith("v$")) {
+                    k = k.substring(2);
+                }
+                if (map.containsKey(k)) {
+                    e.getValue().getWriteMethod().invoke(m, map.get(k));
+                }
+            }
+            return m;
+        } catch (IntrospectionException | IllegalAccessException | InvocationTargetException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    public static <T> T fromMap(Class<T> cls, Map<String, Object> map) {
+        try {
+            Constructor<T> constructor = cls.getConstructor();
+            T t = constructor.newInstance();
+            fromMap(t, map);
+            return t;
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException | InstantiationException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    public static void setId(Object m, String id) {
+        try {
+            BeanInfo beanInfo = Introspector.getBeanInfo(m.getClass());
+            for (PropertyDescriptor pd : beanInfo.getPropertyDescriptors()) {
+                if (pd.getName().equals("id")) {
+                    pd.getWriteMethod().invoke(m, id);
```

Can you add a `idFound` boolean or similar with a check after the `for` loop to make sure the id is available? At the moment this code will silently fail if `id` is not available